src/aws-lambda/mparticle-personalize/mparticle-personalize.js

```diff
@@ -0,0 +1,177 @@
+        
+        for(const param of responseFromSSM.Parameters) {
+            if( param.Name === '/retaildemostore/services_load_balancers/products') {
+                var productsServiceURL = param.Value;
+            } else if (param.Name === '/retaildemostore/webui/mparticle_s2s_api_key') {
+                var mpApiKey = param.Value;
+            } else if (param.Name === '/retaildemostore/webui/mparticle_s2s_secret_key') {
+                var mpApiSecret = param.Value;
+            } else if (param.Name === 'retaildemostore-personalize-event-tracker-id') {
+                var personalizeTrackerID = param.Value; 
+            } else if (param.Name === 'retaildemostore-product-recommendation-campaign-arn') {
+                var personalizeCampaignARN = param.Value;
+            }
+        }
+        
+        // Init mParticle libraries for the function invocation
+        var mpApiInstance = new mParticle.EventsApi(new mParticle.Configuration(mpApiKey, mpApiSecret));
+    } catch (e) {
+        console.log("Error getting SSM parameter for loadbalancer.");
+        console.log(e); 
+        throw e;   
+    }
+
+    for (const record of event.Records) {
+        const payloadString = Buffer.from(record.kinesis.data, 'base64').toString('ascii');
+        const payload = JSON.parse(payloadString);
+        const events = payload.events;
+        var amazonPersonalizeUserId;
+        
+        // First, get the mParticle user ID from the events payload.  In this example, mParticle will send all the events
+        // for a particular user in a batch to this lambda.
+        // retreive the mParticle user id which is available for anonymous and known customer profiles
+        var anonymousID = events[0].data.custom_attributes.mpid.toString();
+        
+        // if the customer profile is known then replace the amazon Personalize User id with the actual
+        // personalize Id captured from the user's profile
+        if(payload.user_attributes && payload.user_attributes.amazonPersonalizeId)
+            amazonPersonalizeUserId = payload.user_attributes.amazonPersonalizeId; 
+        else
+            amazonPersonalizeUserId = anonymousID;
+        // Verify in mParticle's payload if there is a customer id set within the customer profile
+        // this will be used for identity resolution later on within mParticle.
+        var customerId = null;
+        if(payload.user_identities){
+                for (const identityRecord of payload.user_identities)
+                {
+                    if(identityRecord.identity_type==="customer_id")
+                        customerId = identityRecord.identity; 
+                }
+            }
+
+        var params = {
+            sessionId: payload.message_id,
+            userId: amazonPersonalizeUserId,
+            trackingId: personalizeTrackerID,
+            eventList: []
+        };
+
+        for (const e of events) {
```

Comment:
Will the number of events always be <= 10?

Reply:
yes, the number will always be less than 10.